go/v1beta1/example/client/client.go

```diff
@@ -38,6 +187,176 @@ func main() {
+		Details:  &occurenceDeployment,
+	}
+
+	occurenceRequest := pb.CreateOccurrenceRequest{
+		Parent:     "projects/occurrence_example",
+		Occurrence: &occurenceDetails,
+	}
+
+	_, err = client.CreateOccurrence(context.Background(), &occurenceRequest)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	//create a vulnerability occurrence
+	log.Println("create a vulnerability occurrence")
+	packageIssue := vulpb.PackageIssue{
+		AffectedLocation: &vulpb.VulnerabilityLocation{
+			CpeUri:  "7",
+			Package: "a",
+			Version: &packpb.Version{
+				Name:     "v1.1.1",
+				Kind:     3,
+				Revision: "r",
+			},
+		},
+		FixedLocation: &vulpb.VulnerabilityLocation{
+			CpeUri:  "cpe:/o:debian:debian_linux:7",
+			Package: "a",
+			Version: &packpb.Version{
+				Name:     "namestring",
+				Kind:     3,
+				Revision: "1",
+			},
+		},
+	}
+	vulnerability := vulpb.Details{
+		Type: "the type of package",
+		PackageIssue: []*vulpb.PackageIssue{
+			&packageIssue,
+		},
+	}
+	occurrenceVulnerabilityDetails := pb.Occurrence_Vulnerability{
+		Vulnerability: &vulnerability,
 	}
+
+	occurenceDetails = pb.Occurrence{
+		Name: "projects/occurrence_example/occurences/testVulnerabilityOccurrence",
+		Resource: &pb.Resource{
+			Name: "some os",
+			Uri:  "http://someuri",
+		},
+		NoteName: "projects/occurrence_example/notes/testVulnerabilityOccurrence",
+		Kind:     5,
+		Details:  &occurrenceVulnerabilityDetails,
+	}
+
+	occurenceRequest = pb.CreateOccurrenceRequest{
+		Parent:     "projects/occurrence_example",
+		Occurrence: &occurenceDetails,
+	}
+
+	_, err = client.CreateOccurrence(context.Background(), &occurenceRequest)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// now create a build occurrence
+	log.Println("create a build occurrence")
+	build := buildpb.Details{
+		Provenance: &provpb.BuildProvenance{
+			Id:        "build identifier",
+			ProjectId: "some project identifier",
+			Commands:  []*provpb.Command{},
+			BuiltArtifacts: []*provpb.Artifact{
+				&provpb.Artifact{
+					Checksum: "123",
+					Id:       "some identifier for the artifact",
+					Names: []string{
+						"name of the related artifact",
+					},
+				},
+			},
+			CreateTime: &timestamp.Timestamp{
+				Seconds: time.Now().UTC().UnixNano(),
+			},
+			StartTime: &timestamp.Timestamp{
+				Seconds: time.Now().UTC().UnixNano(),
+			},
+			EndTime: &timestamp.Timestamp{
+				Seconds: time.Now().UTC().UnixNano(),
+			},
+			Creator: "User initiating the build",
+			LogsUri: "location of the build logs",
+			SourceProvenance: &provpb.Source{
+				ArtifactStorageSourceUri: "input binary artifacts from this build",
+				Context: &sourcepb.SourceContext{
+					Context: &sourcepb.SourceContext_Git{
+						Git: &sourcepb.GitSourceContext{
+							Url:        "the git repo url",
+							RevisionId: "git commit hash",
+						},
+					},
+				},
+			},
+			TriggerId:      "triggered by code commit 123",
+			BuilderVersion: "some version of the builder",
+		},
+		ProvenanceBytes: "Z3JhZmVhcw==",
```

Comment:
@greendinosaur not sure how to interpret buildProvenanceBytes... per documentation is the json.Marshal of buildProvenance, but is it base64 encoded or not, what is your understanding here?

perhaps @aysylu  or @wkozlik can clarify...

Reply:
Yes, [comment](https://github.com/grafeas/grafeas/blob/master/proto/v1beta1/build.proto#L87) specifies that it's base64 encoded. Does this answer your question?